src/ring_theory/dedekind_domain/selmer_group.lean

```diff
@@ -0,0 +1,107 @@
+/-
+Copyright (c) 2022 David Kurniadi Angdinata. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Kurniadi Angdinata
+-/
+
+import data.zmod.quotient
+import ring_theory.dedekind_domain.adic_valuation
+import ring_theory.localization.basic
+
+/-!
+# Selmer groups
+-/
```

Comment:
Selmer groups are usually subspaces of cohomology groups. What's happening here is that if K is a field then H^1(Gal(K-sep)/K),mu_n) is canonically K^/K^*^n from the long exact sequence and H90.

Reply:
I think there's no harm having these now --- I have explicit definitions of all the maps as mentioned in Zulip. We can always give write down the isomorphisms to the relevant subgroups of H^1 once we have them, and I don't know how long group cohomology will take to eventually make it to mathlib.